scylla/src/transport/session_builder.rs

```diff
@@ -31,8 +43,9 @@ use tracing::warn;
 /// # }
 /// ```
 #[derive(Clone)]
-pub struct SessionBuilder {
+pub struct GenericSessionBuilder<Kind: SessionBuilderKind> {
```

Comment:
There is a potential problem here as the `derive` expands to impl that requires `Clone` to be implemented for `SessionBuilderKind`.  In my opinion, the trait `SessionBuilderKind` should also extend the `Clone` trait and the `DefaultMode` and `CloudMode` structs should also be clonable.

Reply:
There is an open issue about incorrect bounds while using derive macro to implement `Clone`: https://github.com/rust-lang/rust/issues/26925